### ISSUE TITLE
Does not appear to work in `serveronly` mode

### ISSUE THREAD

**Issue by kquinsland** [cbrooker/MMM-Todoist#29]
Hey there :).  Thanks for putting this module together!

I have a question.  Has the ToDoist module been tested with `serveronly`?  (see: https://github.com/MichMich/MagicMirror#server-only)

I have a working MagicMirror install running in a Docker container.  I am able to pull up the magic mirror web-page in a chrome browser.

I am not a JS expert, much less super familiar with `node`, but it appears that there is some form of node helper script that this plugin relies on.  I think this is the case because i see code like this:

```
this.sendSocketNotification("FETCH_TODOIST", this.config);
```
and not a whole lot of code that i would expect to see in a ToDoist API client... but i do see it in the `node_helper.js`

Basically, I see the todoist plugin loading in my browser, but there is no data under it.  I suspect that there is something preventing the helper from loading / fetching the data, but i'm not 100% sure where to being looking / if `serveronly` supports node-helpers.

here's what i see in `STDOUT` when i launch the magic-mirror server:

```
Starting MagicMirror: v2.4.0
Loading config ...
Loading module helpers ...
No helper found for module: alert.
Initializing new module helper ...
Module helper loaded: updatenotification
No helper found for module: clock.
Initializing new module helper ...
Module helper loaded: calendar
No helper found for module: compliments.
No helper found for module: currentweather.
No helper found for module: weatherforecast.
No helper found for module: MMM-CountDown.
All module helpers loaded.
Starting server on port 8080 ...
Server started ...
Connecting socket for: updatenotification
Connecting socket for: calendar
Starting node helper for: calendar
Sockets connected & modules started ...

Ready to go! Please point your browser to: http://0.0.0.0:8080
Create new calendar fetcher for url: http://www.calendarlabs.com/templates/ical/US-Holidays.ics - Interval: 300000
```

*Note:* the _lack_ of `No helper found for module: MMM-Todoist.` implying that `node_helper.js` _was_ found / loaded...

*Note:* the _lack_ of `Starting node helper for: MMM-Todoist`

Thoughts?

**Comment by cbrooker** [comment]
Hi kquinsland,

I've run the module many times in the ServerOnly mode. In fact that's the way I test it while I'm working on it. I haven't however, tested it in the docker deployment option.

I'm looking into it.

Did you run "npm install" in the modules/MMM-Todoist folder?
I also presume that you have created an access token and added it to the config.

For a test try putting ['Inbox'] in the Projects property of the MMM-ToDoist config. If the project ID is bad, it simply won't return anything and looks like it's not working.

**Comment by kquinsland** [comment]
> I haven't however, tested it in the docker deployment option.

Gotcha.  I can't imagine that would change much especially if the browser must initiate the web-socket.  I could be wrong, though.


> Did you run "npm install" in the modules/MMM-Todoist folder?

Yep!  Here's the output

```
# get a bash shell "in" the docker container
sudo docker exec -it magic_mirror /bin/bash

root@7d014dd2dad7:/opt/magic_mirror# cd modules/
root@7d014dd2dad7:/opt/magic_mirror/modules# cd MMM-Todoist/
root@7d014dd2dad7:/opt/magic_mirror/modules/MMM-Todoist# npm install
added 46 packages from 56 contributors and audited 59 packages in 2.044s
```

> For a test try putting ['Inbox'] in the Projects property of the MMM-ToDoist config. If the project ID is bad, it simply won't return anything and looks like it's not working.

Did :(

```
karl@docker-host:/opt/magic_mirror$ cat config/config.js | grep project
                projects: [ "Inbox" ],
```


> I also presume that you have created an access token and added it to the config.

I've tried both the API token approach (settings > integration > API token) and the oAuth > register app > skip flow / temp token approach.

**Comment by kquinsland** [comment]
I got a bit curious and googled frequent problems w/ Docker & WS... lots of noise, but nothing obvious w/r/t a common trap for Docker network configuration... :(


I can see that the MagicMirror page _does_ open a successful web-socket and seems to be using it to push all sorts of interesting things... including the `git status` that is used to push a notification to the frontend when there's an update to the up-stream repos.

I do see the todoist config being pushed and what looks like calendar updates

**Comment by cbrooker** [comment]
Sounds like you did it all. I've been doing some testing and it doesn't look like Todoist has changed anything on the API. The one thing I did notice is that they changed the inbox from having an ID of "Inbox" to an actual integer ID. For example my Inbox Id is now "166564796".

Can you try changing the ID in projects to an actual ID of one of your projects or your inbox?
Open your Todoist inbox in the browser, it will give you a URL like (https://todoist.com/app?lang=en#project%2F166564796). Grab the number after the %2F and put that in your config.

`{
	module: "MMM-Todoist",
	position: "top_right",
	header: "Todoist", 
	config: { 
		accessToken: "",
		maximumEntries: 60,
		updateInterval: 10*60*1000,
		projects: [ 166564796 ],
		fade: false
  }`

See if that gives you anything.

**Comment by kquinsland** [comment]
Yep, i had the project ID for Inbox and one other list already in there. :(

**Comment by cbrooker** [comment]
Hmm, crap. 

Do you mind sharing your MMM-Todist section of the config? (removing your api key, etc)

**Comment by cbrooker** [comment]
Glad it's working for you! Let me know if you have any suggestions. Enjoy your MM.

**Comment by kquinsland** [comment]
You don't have to do this, actually :) . You can get an API token w/o having to get the oAuth flow set up.  Todoist just gives you one to support their SDK and all the other legacy things out there before they supported oAuth auth/scope flow.  Settings, Integration, There's a token at the bottom of the page...

```
Do whatever it needs you to do to get in. Select "Manage App" -> "Create New App"
Give it a name and for "App service URL" just put in "http://localhost".
On the page it takes you to, scroll down to the section "Test Token", click "Create Test token"
It will give you "Your access token", use that in your config.
```
